ovos_tts_plugin_piper/__init__.py

```diff
@@ -308,61 +94,39 @@ def get_model(self, lang=None, voice=None, speaker=None):
-        if voice in self.voice2url:
-            xdg_p = f"{xdg_data_home()}/piper_tts/{voice}"
-            if not os.path.isdir(xdg_p):
-                url = self.voice2url[voice]
-
-                m = url.split("/")[-1]
-                xdg_p = f"{xdg_data_home()}/piper_tts/{m.split('.')[0]}"
-
-                model_file = f"{xdg_p}/{m}"
-                if not os.path.isfile(model_file):
-                    LOG.info(f"downloading piper model: {url}")
-                    os.makedirs(xdg_p, exist_ok=True)
-                    # TODO - streaming download
-                    data = requests.get(url)
-                    with open(model_file, "wb") as f:
-                        f.write(data.content)
+        try:
+            model, model_config = get_voice_files(voice)
+        except VoiceNotFoundError as e:
+            LOG.error(f"Voice files for '{voice}' not found: {e}")
+            raise
 
-                    if url.endswith(".onnx"):
-                        json_data = requests.get(url + '.json')
-                        with open(model_file + '.json', "wb") as f:
-                            f.write(json_data.content)
-                    else:
-                        with tarfile.open(model_file) as file:
-                            file.extractall(xdg_p)
+        return self.get_model(str(model), str(model_config), voice, speaker)
 
-                for f in os.listdir(xdg_p):
-                    if f.endswith(".onnx"):
-                        model = f"{xdg_p}/{f}"
-
-                        with open(model + ".json", "r", encoding="utf-8") as config_file:
-                            config_dict = json.load(config_file)
-
-                        engine = PiperVoice(
-                            config=PiperConfig.from_dict(config_dict),
-                            session=onnxruntime.InferenceSession(
-                                str(model),
-                                sess_options=onnxruntime.SessionOptions(),
-                                providers=["CPUExecutionProvider"]
-                                if not self.use_cuda
-                                else ["CUDAExecutionProvider"],
-                            ),
-                        )
-
-                        LOG.debug(f"loaded model: {model}")
-                        PiperTTSPlugin.engines[voice] = engine
-                        return engine, speaker
-                else:
-                    raise FileNotFoundError("onnx model not found")
-        else:
-            raise ValueError(f"invalid voice: {voice}")
+    def get_model(self, model: str, model_config: str,
+                  voice: str = None, speaker=0):
+        voice = voice or self.voice
+        with open(model_config, "r", encoding="utf-8") as config_file:
+            config_dict = json.load(config_file)
+
+        engine = PiperVoice(
+            config=PiperConfig.from_dict(config_dict),
+            session=onnxruntime.InferenceSession(
+                str(model),
```

Comment:
You've already defined model as a string in the method signature. Is there a use case where it might come in as a different type?

Reply:
it can be a Path i think, not sure why the cast to str was there so i didnt change that